File: Chapter1/S1.4/Python/exercise04.py
import pandas as pd

ages = [22, 22, 22, 22, 23, 23, 24, 24, 24, 24, 25, 25, 26, 26, 26, 26, 27, 28,\
        35, 40]
        
age_freq = pd.Series(ages).value_counts()
age_freq_norm = pd.Series(ages).value_counts(normalize=True)

# Build table with the frequency, percentage, and cumulative percentage
df = pd.DataFrame()
df["Age Frequency"] = age_freq
df["Relative"] = age_freq_norm
df = df.sort_index()
df["Relative Cumulative"] = df["Relative"].cumsum()

print("Age distribution frequency")
print(df)

# Exercise asks us to remove atipical ages: 35 and 40 in this case
ages_no_outlier = [x for x in ages if x < 30]
age_no_outlier_freq = pd.Series(ages_no_outlier).value_counts()
age_no_outlier_freq_norm = \
                 pd.Series(ages_no_outlier).value_counts(normalize=True)

df_no_outlier = pd.DataFrame()

df_no_outlier["Age Frequency"] = age_no_outlier_freq
df_no_outlier["Relative"] = age_no_outlier_freq_norm
df_no_outlier = df_no_outlier.sort_index()
df_no_outlier["Relative Cumulative"] = df_no_outlier["Relative"].cumsum()

print()
print("Age distribution frequency without outliers")
print(df_no_outlier)

File: Chapter1/S1.4/R/exercise04.r
ages <- c(22, 22, 22, 22, 23, 23, 24, 24, 24, 24, 25, 25, 26, 26, 26, 26, 27, 28, 35, 40)

# Build table with the frequency, percentage, and cumulative percentage
age_freq = table(ages)
age_freq_norm = age_freq / length(ages)
age_freq_norm_cum = cumsum(age_freq_norm)

df = cbind(age_freq, age_freq_norm, age_freq_norm_cum)
colnames(df) <- c("Age Frequency", "Relative", "Relative Cumulative")

print("Age distribution frequency")
print(df)

# Exercise asks us to remove atipical ages: 35 and 40 in this case
ages_no_outlier = ages[ages < 30]
age_no_outlier_freq = table(ages_no_outlier)
age_no_outlier_freq_norm = age_no_outlier_freq / length(ages_no_outlier)
age_no_outlier_freq_norm_cum = cumsum(age_no_outlier_freq_norm)

df_no_outlier = cbind(age_no_outlier_freq, age_no_outlier_freq_norm, age_no_outlier_freq_norm_cum)
colnames(df_no_outlier) <- c("Age Frequency", "Relative", "Relative Cumulative")

print("")
print("Age distribution frequency without outliers")
print(df_no_outlier)
